src/common/Exporter.js

```diff
@@ -0,0 +1,80 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define([], factory);
+    } else {
+        root.common_Exporter = factory();
+    }
+}(this, function () {
+    function Exporter(widget) {
+        this._widget = widget;
+    }
+
+    Exporter.prototype.formatData = function (_) {
+        var formattedData;
+        switch(_) {
+            case "TSV":
+                formattedData = this._widget._db.tsv();
+                break;
+            case "JSON":
+                formattedData = this._widget._db.json();
+                break;
+            default:
+                formattedData = this._widget._db.csv();
+        }
+        return formattedData;
+    };
+
+
+    Exporter.prototype.downloadData = function (format) {
+        var data = this.formatData(format);
+
+        var a = document.createElement('a');
+
+        var currentdate = new Date(); 
+        var now = currentdate.getDate() + "."+ 
+            (currentdate.getMonth()+1)  + "." + 
+            currentdate.getFullYear() + "_"  + 
+            currentdate.getHours() + ":"  + 
+            currentdate.getMinutes() + ":" + 
+            currentdate.getSeconds()
+        ;
+        var filename = this._widget.classID() + "_" + now + "." + format.toLowerCase();
+        
+        var mimeType = "";
+        switch (format) {
+            case "TSV":
+                mimeType = "text/tab-seperated-values";
+                break;
+            case "JSON":
+                mimeType = "application/json";
+                break;
+            default:
+                mimeType = "text/csv";
+        }
+
+        if (navigator.msSaveBlob) { // IE10+
+            return navigator.msSaveBlob(new Blob([data], { type: mimeType }), filename);
+        } else if ('download' in a) { //html 5
+            a.href = 'data:' + mimeType + ',' + encodeURIComponent(data);
+            a.setAttribute('download', filename);
+            document.body.appendChild(a);
+            setTimeout(function() {
+                a.click();
+                document.body.removeChild(a);
+            }, 10);
+            return true;
+        } else { //old chrome and FF:
+            var frame = document.createElement('iframe');
+            document.body.appendChild(frame);
+            frame.src = 'data:' + mimeType + ',' + encodeURIComponent(data);
+
+            setTimeout(function() {
+                document.body.removeChild(frame);
+            }, 100);
+            return true;
+            }
```

Comment:
This should work on everything IE 9+.  I have IE 10, and used the dev tools to emulate IE9 and it works (but that is not always valid).